server/src/internalClusterTest/java/org/elasticsearch/search/sort/FieldSortIT.java

```diff
@@ -868,6 +868,142 @@ public void testSortMissingStrings() throws IOException {
+                    .addSort(SortBuilders.fieldSort("mydate").order(SortOrder.ASC).missing("_first").setFormat(format))
+                    .get();
+                assertHitsInOrder(searchResponse, new String[] { "3", "1", "2" });
+
+                searchResponse = client().prepareSearch(index)
+                    .addSort(SortBuilders.fieldSort("mydate").order(SortOrder.DESC).setFormat(format))
+                    .get();
+                assertHitsInOrder(searchResponse, new String[] { "2", "1", "3" });
+
+                searchResponse = client().prepareSearch(index)
+                    .addSort(SortBuilders.fieldSort("mydate").order(SortOrder.DESC).missing("_first").setFormat(format))
+                    .get();
+                assertHitsInOrder(searchResponse, new String[] { "3", "2", "1" });
+            }
+        }
+    }
+
+    /**
+     * Sort across two indices with both "date" and "date_nanos" type using "numeric_type" set to "date_nanos"
+     */
+    public void testSortMissingDatesMixedTypes() throws IOException {
+        for (String type : List.of("date", "date_nanos")) {
+            String index = "test_" + type;
+            assertAcked(
+                prepareCreate(index).setMapping(
+                    XContentFactory.jsonBuilder()
+                        .startObject()
+                        .startObject("_doc")
+                        .startObject("properties")
+                        .startObject("mydate")
+                        .field("type", type)
+                        .endObject()
+                        .endObject()
+                        .endObject()
+                        .endObject()
+                )
+            );
+
+        }
+        ensureGreen();
+
+        client().prepareIndex("test_date").setId("1").setSource("mydate", "2021-01-01").get();
+        client().prepareIndex("test_date").setId("2").setSource("mydate", "2021-02-01").get();
+        client().prepareIndex("test_date").setId("3").setSource("other_field", "value").get();
+        client().prepareIndex("test_date_nanos").setId("4").setSource("mydate", "2021-03-01").get();
+        client().prepareIndex("test_date_nanos").setId("5").setSource("mydate", "2021-04-01").get();
+        client().prepareIndex("test_date_nanos").setId("6").setSource("other_field", "value").get();
+        refresh();
+
+            for (boolean withFormat : List.of(true, false)) {
+                String format = null;
+                if (withFormat) {
+                    format = "strict_date_optional_time_nanos";
+                }
+
+                String index = "test*";
+                SearchResponse searchResponse = client().prepareSearch(index)
+                    .addSort(SortBuilders.fieldSort("mydate").order(SortOrder.ASC).setFormat(format).setNumericType("date_nanos"))
+                    .get();
+                assertHitsInOrder(searchResponse, new String[] { "1", "2", "4", "5", "3", "6" });
```

Comment:
A tiebreaker should be used to ensure that `3` is always sorted before `6` (same value otherwise).